features/machine/machine.feature

```diff
@@ -171,3 +171,31 @@ Feature: Machine features testing
       | role=master:NoExecute |
       | role=master:NoSchedule|
 
+  # @author zhsun@redhat.com
+  @admin
+  @destructive
+  Scenario Outline: Required configuration should be added to the ProviderSpec to enable spot instances
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+
+    Given I store the number of machines in the :num_to_restore clipboard
+    And admin ensures node number is restored to "<%= cb.num_to_restore %>" after scenario
+
+    #Create a spot machineset
+    Given I use the "openshift-machine-api" project
+    Given I create a spot instance machineset named "<machineset_name>" on <iaas_type>
+    And evaluation of `machine_set.machines.first.node_name` is stored in the :noderef_name clipboard
+    And evaluation of `machine_set.machines.first.name` is stored in the :machine_name clipboard
+    
+    #Check machine and node were labelled as an `interruptible-instance`
+     When I run the :describe admin command with:
+      | resource | machine                |
+      | name     | <%= cb.machine_name %> |
+    Then the step should succeed
+    And the output should match "machine.openshift.io/interruptible-instance"
+    And label "machine.openshift.io/interruptible-instance=" is added to the "<%= cb.noderef_name %>" node
```

Comment:
What's the purpose of adding a label to a node at the last step ?

Reply:
All Machines will either be Spot or On-Demand,they will be labeled  as interruptible if they are spot instance. A Termination Pod DaemonSet would be deployed and will select Nodes which are labelled as spot instances. This is test case [OCP-29389](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-29389).

Reply:
Perhaps you want to check node has the label ? Or you want to add label to the node ?

Reply:
Check the node has this label

Reply:
The step is adding label to node.

Reply:
Thanks, have updated.